src/api/github/pullRequest.ts

```diff
@@ -0,0 +1,76 @@
+import { Octokit } from '@octokit/rest';
+
+import * as Git from '../git';
+
+import { apiTokenFromEnvironment } from './environment';
+
+interface GetPullRequestParameters {
+  /**
+   * A preconstructed Octokit client to interact with GitHub's APIs.
+   *
+   * A `GITHUB_API_TOKEN` or `GITHUB_TOKEN` with write permissions must be
+   * present on the environment if this is not provided.
+   */
+  client?: Octokit;
+
+  env?: Record<string, string | undefined>;
+}
+
+interface PullRequest {
+  number: number;
+}
+
+/**
+ * Gets the number of the current pull request.
+ *
+ * This tries to extract the pull request from common CI environment variables,
+ * and falls back to querying the GitHub Repos API for the latest pull request
+ * associated with the head commit. An error is thrown if there are no
+ * associated pull requests or if they are all closed.
+ */
+export const getPullRequest = async (
```

Comment:
question: Would it be worth exposing any of the other fields for the PR? I guess if I'm calling `getPullRequest` I would be expecting more than a simple number?

Reply:
Yeah good point. I wanted to keep the API small until there was a need for extra fields, but I figure we can add a few more non-controversial ones.

Reply:
On second thought, I'll scope this to `getPullRequestNumber`. The nicety here is that we can drive the function purely off of environment variables on the happy path.